page.go

```diff
@@ -69,70 +72,40 @@ func (p page) Marshal() ([]byte, error) {
 
 	buffer := new(bytes.Buffer)
 
-	// write payloadSize and payload
-	err1 := binary.Write(buffer, binary.LittleEndian, uint64(len(p.payload)))
-	_, err2 := buffer.Write(p.payload)
+	// write checksum, pageStatus, transactionNumber and nextPage
+	_, err1 := buffer.Write(p.transactionChecksum[:])
+	err2 := binary.Write(buffer, binary.LittleEndian, p.pageStatus)
+	if p.pageStatus == pageStatusInvalid {
+		panic(errors.New("Sanity check failed. Page was marshalled with invalid PageStatus"))
+	}
 
-	// write pageStatus, transactionNumber, nextPage and checksum
-	err3 := binary.Write(buffer, binary.LittleEndian, p.pageStatus)
-	err4 := binary.Write(buffer, binary.LittleEndian, p.transactionNumber)
-	err5 := binary.Write(buffer, binary.LittleEndian, nextPage)
-	_, err6 := buffer.Write(p.transactionChecksum[:])
+	err3 := binary.Write(buffer, binary.LittleEndian, p.transactionNumber)
+	err4 := binary.Write(buffer, binary.LittleEndian, nextPage)
```

Comment:
it's declared a bit earlier, but I'd probably call the `nextPage` variable `nextPagePosition`, since in the actual struct it is the actual next page, but in this struct it's just a location on disk.